crates/router/src/core/payment_methods/cards.rs

```diff
@@ -538,9 +621,17 @@ fn filter_payment_currency_based(
     pm: &api::ListPaymentMethod,
 ) -> bool {
     payment_intent.currency.map_or(true, |currency| {
-        pm.accepted_currencies
-            .clone()
-            .map_or(true, |ac| ac.contains(&currency.foreign_into()))
+        pm.accepted_currencies.clone().map_or(true, |ac| {
```

Comment:
You can  use `as_ref()` instead of `clone()` here.

Reply:
yes